Typo in SparseMatrix.hpp - unorderd_map

Attempting to build with a modern GCC I get 

```
/usr/bin/g++ -c -DHPCG_NO_MPI -I./src -I./src/GCC_OMP  -O3 -ffast-math -ftree-vectorize -ftree-vectorizer-verbose=0 -fopenmp -I../src ../src/main.cpp -o src/main.o
In file included from ../src/GenerateProblem.hpp:17:0,
                 from ../src/main.cpp:42:
../src/SparseMatrix.hpp:36:31: error: ‘unorderd_map’ in namespace ‘std’ does not name a template type
 using GlobalToLocalMap = std::unorderd_map< global_int_t, local_int_t >;
                               ^~~~~~~~~~~~
```

Spelling it `unordered_map` allows the build to complete

--- a/src/SparseMatrix.hpp
+++ b/src/SparseMatrix.hpp
@@ -33,7 +33,7 @@ typedef std::map< global_int_t, local_int_t > GlobalToLocalMap;
 #else
 // for C++11 or greater
 #include <unordered_map>
-using GlobalToLocalMap = std::unorderd_map< global_int_t, local_int_t >;
+using GlobalToLocalMap = std::unordered_map< global_int_t, local_int_t >;
 #endif
 
 struct SparseMatrix_STRUCT {

